scripts/un/energy/download.py

```diff
@@ -123,15 +102,28 @@ def download_energy_dataset(
         end_year = year_batch[-1]
         years_str = ','.join(year_batch)
         output = f'{output_path}-{energy_dataset}-{start_year}-{end_year}'
-        print('Downloading UNData energy dataset: ',
-              f'{energy_dataset} from {start_year} to {end_year}',
-              flush=True)
+        logging.info(
+            f'Downloading UNData energy dataset: {energy_dataset} from {start_year} to {end_year}'
+        )
         download_url = _DOWNLOAD_URL.format(energy_code=energy_dataset,
                                             years=years_str)
-        downloaded_files = download_zip_file(download_url, output)
-        if len(downloaded_files) == 0:
-            errors += 1
-        output_files.extend(downloaded_files)
+        download_successful = download_file(url=download_url,
+                                            output_folder=output,
+                                            unzip=True,
+                                            headers=None,
+                                            tries=3,
+                                            delay=5,
+                                            backoff=2)
+        if download_successful:
+            logging.info(f"Download of '{download_url}' completed.")
+            output_files = []
+            for f in os.listdir(output):
+                output_files.append(os.path.join(output, f))
+                if len(output_files) == 0:
```

Comment:
optional nit: can do something like 
if not output_files

Also, what is errors counting? It looks like output_files is getting re initialized every instance of the loop - should there instead be a separate variable for the per-url files?

Reply:
Could you clarify or add a comment for what the errors variable is counting? Is it whether output is empty? (If so it won't get checked in the loop) Alternatively if the first combination of downloads is not empty, output_files won't be, so later combinations won't have the same check